### ISSUE TITLE
The extension does not seem to work well

### ISSUE THREAD

**Issue by ljleb** [xingqian2018/sd-webui-prompt-free-diffusion#2]
Hi, I ran some tests with the extension after downloading all 3 seecoder models. I am unable to get any good results with any of them, no matter what setting I use. The webui is only able to generate something intelligible when the extension dropdown is disabled.

<details>

<summary>Example settings</summary>

![image](https://github.com/xingqian2018/sd-webui-prompt-free-diffusion/assets/32277961/04b66c9d-514c-4700-a6f6-35ee2122a0bd)

</details>

<details>

<summary>Input image</summary>

![000000](https://github.com/xingqian2018/sd-webui-prompt-free-diffusion/assets/32277961/d95ce73b-b242-4fe0-9764-695d078d87ab)

</details>

When I restart the webui and load a previously generated image from the PNG Info tab, then press generate, I get a different scrambled mess every time for the first generation. Then, all follow-up generations (without restarting the webui) from the 2nd generation onward are all the same:

<details>

<summary>Sampling result</summary>

![47917-1860717067-masterpiece, best quality, hires,_cute 1girl looking at viewer_](https://github.com/xingqian2018/sd-webui-prompt-free-diffusion/assets/32277961/a0a4f88e-3adb-4656-88be-65203e0b1606)

</details>

(note that this image is not the same as the first generation that you can find in the settings image above. the extension consistently generates this image after the first generation)

<details>

<summary>Logs</summary>

```
Python 3.10.9 (tags/v3.10.9:1dd9be6, Dec  6 2022, 20:01:21) [MSC v.1934 64 bit (AMD64)]
Version: v1.0.0-pre-1578-g394ffa7b
Commit hash: 394ffa7b0a7fff3ec484bcd084e673a8b301ccc8
Installing requirements
Launching Web UI with arguments: --api --api-log --allow-code --ckpt-dir E:\sd\models\Stable-diffusion --embeddings-dir E:\sd\models\embeddings --vae-dir E:\sd\models\VAE --lora-dir E:\sd\models\Lora --esrgan-models-path E:\sd\models\ESRGAN --realesrgan-models-path E:\sd\models\RealESRGAN --controlnet-dir E:\sd\models\ControlNet --controlnet-annotator-models-path E:\sd\models
D:\src\stable-diffusion-webui\venv\lib\site-packages\pkg_resources\__init__.py:123: PkgResourcesDeprecationWarning: rch is an invalid version and will not be supported in a future release
  warnings.warn(
No module 'xformers'. Proceeding without it.
2023-07-02 22:21:16,929 - PFD - INFO - Prompt-Free-Diffusion v1.0.1
2023-07-02 22:21:17,605 - PFD - INFO - Prompt-Free-Diffusion v1.0.1
Loading weights [de329f31ad] from E:\sd\models\Stable-diffusion\silica\models\original\Silicon29-dark.safetensors
Running on local URL:  http://127.0.0.1:7860
To create a public link, set `share=True` in `launch()`.
Startup time: 8.6s (import torch: 2.1s, import gradio: 1.2s, import ldm: 0.6s, other imports: 1.0s, setup codeformer: 0.1s, list SD models: 0.5s, load scripts: 2.0s, create ui: 0.6s, gradio launch: 0.3s, add APIs: 0.2s).
Creating model from config: D:\src\stable-diffusion-webui\configs\v1-inference.yaml
LatentDiffusion: Running in eps-prediction mode
DiffusionWrapper has 859.52 M params.
Loading VAE weights specified in settings: E:\sd\models\VAE\kl-f8-anime2.vae.pt
Applying attention optimization: Doggettx... done.
Model loaded in 6.8s (load weights from disk: 1.1s, create model: 0.7s, apply weights to model: 0.6s, apply half(): 0.9s, load VAE: 0.6s, move model to device: 0.9s, load textual inversion embeddings: 1.9s, calculate empty prompt: 0.1s).
2023-07-02 22:21:28,903 - PFD - INFO - Loading model: seecoder-v1-0 [936e345d]
2023-07-02 22:21:28,956 - PFD - INFO - Loaded state_dict from [E:\sd\models\SeeCoder\seecoder-v1-0.safetensors]
2023-07-02 22:21:28,956 - PFD - INFO - Loading config: D:\src\stable-diffusion-webui\extensions\sd-webui-prompt-free-diffusion\models\seecoder-v1-0.yaml
preload_extensions_git_metadata for 86 extensions took 9.69s
2023-07-02 22:21:30,781 - PFD - INFO - seecoder-v1-0 [936e345d] loaded.
  3%|â–Ž         | 1/30 [00:00<00:21,  1.35it/s]
Total progress:   0%|          | 0/30 [00:00<?, ?it/s]
 13%|â–ˆâ–Ž        | 4/30 [00:01<00:06,  4.25it/s]
 20%|â–ˆâ–ˆ        | 6/30 [00:01<00:04,  5.88it/s]
Total progress:  20%|â–ˆâ–ˆ        | 6/30 [00:00<00:02,  8.00it/s]
 27%|â–ˆâ–ˆâ–‹       | 8/30 [00:01<00:03,  6.70it/s]
Total progress:  27%|â–ˆâ–ˆâ–‹       | 8/30 [00:00<00:02,  7.80it/s]
 33%|â–ˆâ–ˆâ–ˆâ–Ž      | 10/30 [00:01<00:02,  7.21it/s]
Total progress:  33%|â–ˆâ–ˆâ–ˆâ–Ž      | 10/30 [00:01<00:02,  7.78it/s]
 40%|â–ˆâ–ˆâ–ˆâ–ˆ      | 12/30 [00:02<00:02,  7.23it/s]
Total progress:  40%|â–ˆâ–ˆâ–ˆâ–ˆ      | 12/30 [00:01<00:02,  7.52it/s]
 47%|â–ˆâ–ˆâ–ˆâ–ˆâ–‹     | 14/30 [00:02<00:02,  7.67it/s]
Total progress:  47%|â–ˆâ–ˆâ–ˆâ–ˆâ–‹     | 14/30 [00:01<00:02,  7.80it/s]
 53%|â–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–Ž    | 16/30 [00:02<00:01,  7.59it/s]
Total progress:  53%|â–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–Ž    | 16/30 [00:02<00:01,  7.68it/s]
 60%|â–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆ    | 18/30 [00:03<00:01,  7.57it/s]
Total progress:  60%|â–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆ    | 18/30 [00:02<00:01,  7.59it/s]
 67%|â–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–‹   | 20/30 [00:03<00:01,  7.14it/s]
Total progress:  67%|â–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–‹   | 20/30 [00:02<00:01,  7.17it/s]
 73%|â–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–Ž  | 22/30 [00:03<00:01,  7.37it/s]
Total progress:  73%|â–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–Ž  | 22/30 [00:02<00:01,  7.38it/s]
 80%|â–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆ  | 24/30 [00:03<00:00,  7.36it/s]
Total progress:  80%|â–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆ  | 24/30 [00:03<00:00,  7.35it/s]
 87%|â–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–‹ | 26/30 [00:04<00:00,  7.28it/s]
Total progress:  87%|â–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–‹ | 26/30 [00:03<00:00,  7.29it/s]
 93%|â–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–Ž| 28/30 [00:04<00:00,  7.00it/s]
Total progress:  93%|â–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–Ž| 28/30 [00:03<00:00,  7.00it/s]
100%|â–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆ| 30/30 [00:04<00:00,  6.40it/s]
Total progress: 100%|â–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆ| 30/30 [00:03<00:00,  7.43it/s]
Total progress: 100%|â–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆ| 30/30 [00:04<00:00,  6.94it/s]
{"prompt": "", "all_prompts": [""], "negative_prompt": "", "all_negative_prompts": [""], "seed": 1860717067, "all_seeds": [1860717067], "subseed": 409547144, "all_subseeds": [409547144], "subseed_strength": 0, "width": 512, "height": 512, "sampler_name": "DPM++ 2M Karras", "cfg_scale": 2, "steps": 30, "batch_size": 1, "restore_faces": false, "face_restoration_model": null, "sd_model_hash": "de329f31ad", "seed_resize_from_w": 0, "seed_resize_from_h": 0, "denoising_strength": null, "extra_generation_params": {"ControlNet": "model: seecoder-v1-0 [936e345d], weight: 1"}, "index_of_first_image": 0, "infotexts": ["Steps: 30, Sampler: DPM++ 2M Karras, CFG scale: 2, Seed: 1860717067, Size: 512x512, Model hash: de329f31ad, Model: silica_models_original_Silicon29-dark, ControlNet: \"model: seecoder-v1-0 [936e345d], weight: 1\", Version: v1.0.0-pre-1578-g394ffa7b"], "styles": [], "job_timestamp": "20230702222128", "clip_skip": 1, "is_using_inpainting_conditioning": false}
```

</details>

Is there any other piece of information I can provide? I want to believe this is a problem with the extension code, and not with the technique described in the paper.

Note that despite my logs saying I am on `Version: v1.0.0-pre-1578-g394ffa7b`, I pulled the latest version of the webui (v1.4.0) before running these tests. I have disabled every other extension (apart from the builtin ones) and completely restarted the webui.

**Comment by xingqian2018** [comment]
For anime, I suggest you use seecoder-anime model. Also, it would be favorable to include controlnet with structural control input such as canny edge or pose. 

In a high-level view, one-shot image-variation with ID preservation is hard and ongoing research as well. SeeCoder is a solution to that but it still has its limitations.

**Comment by ljleb** [comment]
Putting the reproduction of the paper aside, do you know what could be causing this issue? (from my comment above)

> When I restart the webui and load a previously generated image from the PNG Info tab, then press generate, I get a different scrambled mess every time for the first generation. Then, all follow-up generations (without restarting the webui) from the 2nd generation onward are all the same:

I would think seed generation should be deterministic, but it isn't for the first generation after starting thewebui.

**Comment by ljleb** [comment]
Are you able to reproduce the results of the paper on your side? Even with a realistic model, I seem to be getting better results by disabling the extension nearly every time. Of course I could simply be biased in my qualitative judgements, but I want to verify this really isn't a problem caused by the specific implementation of the extension.

**Comment by maepopi** [comment]
Hello! Allow me to jump in here, I do also struggle to reproduce the results in the paper, and I'm wondering whether I'm doing anything wrong.

I'm trying to reproduce the result of the astronaut on the horse.

![image](https://github.com/user-attachments/assets/90424508-6869-4486-9b35-3920621af688)

I'm using the canny as controlnet

![image](https://github.com/user-attachments/assets/12bac426-2e00-44aa-8fe0-bf1700f4a4bd)

and here's the result

![image](https://github.com/user-attachments/assets/212d6207-3ee5-4029-b376-694e022f474d)

using the "balance" option allows to get closer to the paper result but it still remains pretty far. I think the canny is way too pronounced, maybe that's coming from A111? 

![image](https://github.com/user-attachments/assets/604ca526-864f-4f84-a125-770569b31201)


Thanks!
